### DISCUSSION TITLE
Feedback on new Runs page

### DISCUSSION THREAD

**Discussion by jamiedemaria** [dagster-io/dagster#24898] [Community]
In dagster version 1.8.10 we released an experimental new version of the Runs page. The new page incorporates backfills into the Runs page to simplify the experience of navigating the UI and finding the actions launched by dagster. 

### What’s changed

The Runs page now contains a row for each backfill launched in dagster. Runs that are executed by the backfill are hidden from the Runs page by default, and can be found by clicking into a backfill or selecting the “Show runs within backfills” checkbox. 

### Motivation

The details page for a backfill contains useful information about the status of the backfill process. However, backfills are hard to find in the UI (nested in the Overview page). Additionally,  knowing to look for this page requires that a user knows how dagster will execute their request (as a backfill or as a single run). By including backfills in the Runs page, the information about the backfill is more available in the UI.

If a large backfill is launched, the Runs page quickly becomes overrun by the runs the backfill launches. This can make it difficult to see the other runs dagster is executing. By hiding runs that are part of a backfill by default, it is easier to see the top-level actions that dagster is executing.

### How to enable the new Runs page

To enable the new Runs page, switch the “New Runs page” user setting to on. 
<img width="484" alt="Screenshot 2024-09-30 at 12 13 11 PM" src="https://github.com/user-attachments/assets/2f36bb72-6f1d-43eb-be2d-57fae295c48d">

### Request for feedback

Please tell us about your experience using the new Runs page. We’re actively looking for feedback about how to make the new page more useful, and will be continuing to make improvements over the coming weeks.

**Comment by Dduran1991** [comment]
Its cool , but I dont see any tag that says that its a backfill, looks like another run from external view
![backfills](https://github.com/user-attachments/assets/e60d048f-4853-4a81-b584-37ee2ff03a38)

**Reply by jamiedemaria** [reply]
hey @Dduran1991  thanks for the feedback. We're experimenting with finding the right balance between making making backfills and runs look too similar or too different so we'll discuss it on our end as we decide what changes we want to make for this view. Right now, the distinguishing feature is the circle icon on the far left.

**Comment by HynekBlaha** [comment]
Hello, I would like to report that if `Experimental features -> New Runs page` is enabled the Runs tab timeouts for me.
We are running Dagster in K8s (20~ code deployments for 1,5~ year).
<img width="1789" alt="Screenshot 2024-10-08 at 19 02 20" src="https://github.com/user-attachments/assets/600a45b2-06aa-451f-a9af-1b635429276a">
<img width="1285" alt="Screenshot 2024-10-08 at 19 02 27" src="https://github.com/user-attachments/assets/26f85fb0-6b17-4d36-9a67-8b6ea7cf45b7">

**Reply by jamiedemaria** [reply]
thanks for reporting this - we'll look into it

**Reply by jamiedemaria** [reply]
hey @HynekBlaha we're looking into this, but could use some information from you - what version of dagster are you running? what database are you using? what's the latest alembic migration that you've run (you can run `select * from alembic_version` to find out)

**Reply by HynekBlaha** [reply]
Hi @jamiedemaria, here are the answers. :)

what version of dagster are you running
> 1.8.10

what database are you using
> Postgres 15.7 (General Purpose, D4ds_v5, 4 vCores, 16 GiB RAM, 512 GiB storage)

select * from alembic_version
> 5771160a95ad

**Reply by jamiedemaria** [reply]
ok - we added some additional indices a few months ago that should help with this query. If you are able, can you migrate your database and see if that improves performance?

**Reply by HynekBlaha** [reply]
Hello @jamiedemaria,
I was able to run migrations on out Dagster STAGING instance.
Here is the job output:
```
Storage schema state before migration:
--------------------------------------
schema:
  event_log_storage:
    current: 5771160a95ad
    latest: 284a732df317
  run_storage:
    current: 5771160a95ad
    latest: 284a732df317
  schedule_storage:
    current: 5771160a95ad
    latest: 284a732df317


Running migration:
------------------
Updating run storage...
Skipping already applied data migration: run_partitions
Skipping already applied data migration: run_repo_label_tags
Skipping already applied data migration: bulk_action_types
Updating event storage...
Skipping already applied data migration: asset_key_index_columns
Updating schedule storage...
Skipping already applied migration: schedule_jobs_selector_id

Storage schema state after migration:
-------------------------------------
schema:
  event_log_storage:
    current: 284a732df317
    latest: 284a732df317
  run_storage:
    current: 284a732df317
    latest: 284a732df317
  schedule_storage:
    current: 284a732df317
    latest: 284a732df317
```

However, even with applied alembic migrations, it takes long time to fetch the runs on "Experimental Run Page" (/runs).
<img width="612" alt="Screenshot 2024-10-21 at 20 51 18" src="https://github.com/user-attachments/assets/afa01d6a-75eb-4c6b-8161-362707d9c9fd">.

The "Overview Page [24h]" (/overview/activity/timeline) is much more responsive.
<img width="609" alt="Screenshot 2024-10-21 at 20 56 57" src="https://github.com/user-attachments/assets/ce3fb546-6f6f-4807-a9b2-564b2eefdda8">

Note: We have ±30 code deployments.

**Reply by HynekBlaha** [reply]
Hello @jamiedemaria, I have added `explain analyze` queries to provide more insight into our troubles.
Could you please take a look at it? Any recommendations are very appreciated.
https://github.com/dagster-io/dagster/issues/25659
Thank you, Hynek

**Comment by brennan-wall-fgp** [comment]
The complete removal of the "Backfills" page is missed.  That was useful for quickly checking on the status of a backfill or seeing how long your previous backfill took.

- If you are committed to the removal of the Backfills page it seems like there should be a filter for "is backfill" on the Runs page
- 2nding that the icon difference between runs and backfills wasn't obvious

**Reply by jamiedemaria** [reply]
Hey @brennan-wall-fgp - the backfill page is back for the time being, and we are working on some ways to distinguish backfills more in the new page

**Comment by brunobento-delfos** [comment]
Having the partitions progress bar for backfills on the Run Page would be useful for us. It's kind of tedious to enter one backfill at a time to check its progress

**Reply by HynekBlaha** [reply]
We were just discussing this on todays stand-up...
We are running Dagster OSS on Kubernetes. Since the Backfill daemon enqueues new runs one by one, it is very critical for us to quickly investigate who is blocking the queue by submitting a large backfill (in the thousands of partitions).

Before Dagster 1.8, the backfill page showed both Backfill status (enqueing) and backfill progress bar.
In Dagster 1.8, the progress bar was removed, now we have to go inside a backfill to see what is its progress.
In Dagster 1.9, there is a warning about Backfill page being deprecated and Backfills will be showed on the Runs page.

I understand changes can be done in order to increase performace.
But I would like to emphasize that the general Backfill overview with status/progress is a crucial to have.
cc: @jamiedemaria
